f8a_report/report_helper.py

```diff
@@ -540,66 +538,75 @@ def normalize_ingestion_data(self, start_date, end_date, ingestion_data, frequen
                 'generated_on': dt.now().isoformat('T')
             },
             'ingestion_summary': {},
-            'ingestion_details': []
+            'ingestion_details_v2': {}
         }
 
         all_deps_count = {'all': 0, 'npm': 0, 'maven': 0, 'python': 0}
         failed_deps_count = {'all': 0, 'npm': 0, 'maven': 0, 'python': 0}
-        all_epv_list = {'npm': [], 'maven': [], 'python': []}
-        failed_epv_list = {'npm': [], 'maven': [], 'python': []}
+        all_epv_list_v2 = {'npm': {}, 'maven': {}, 'python': {}}
 
-        # marshalling the total ingested epv data according to the ecosystems
-        epv_data = ingestion_data['EPV_INGESTION_DATA']
-        epv_data = json.loads(epv_data)
-        for data in epv_data:
+        # Graph Availability validation
+        success_epv_data = ingestion_data['EPV_GRAPH_SUCCESS_DATA']
+        success_epv_data = json.loads(success_epv_data)
+        graph_input = []
+        for data in success_epv_data:
             all_deps_count['all'] = all_deps_count['all'] + 1
-            if data[0] == 'maven':
-                all_deps_count['maven'] = all_deps_count['maven'] + 1
-                all_epv_list['maven'].append(data[1] + '::' + data[2])
-            elif data[0] == 'npm':
-                all_deps_count['npm'] = all_deps_count['npm'] + 1
-                all_epv_list['npm'].append(data[1] + '::' + data[2])
-            elif data[0] == 'python':
-                all_deps_count['python'] = all_deps_count['python'] + 1
-                all_epv_list['python'].append(data[1] + '::' + data[2])
-            else:
-                continue
+            all_deps_count[data[0]] = all_deps_count[data[0]] + 1
+            graph_template = {
+                'ecosystem': data[0],
+                'name': data[1],
+                'version': data[2]
+            }
+            graph_input.append(graph_template)
+
+        graph_output = generate_report_for_unknown_epvs(graph_input)
+        for attributes, values in graph_output.items():
+            versn_template = {}
+            epv_arr = attributes.split('@')
+            all_epv_list_v2[epv_arr[0]][epv_arr[1]] = {}
+            all_epv_list_v2[epv_arr[0]][epv_arr[1]]['package_known'] = values
+            all_epv_list_v2[epv_arr[0]][epv_arr[1]]['versions'] = []
+            versn_template['version'] = epv_arr[2]
+            versn_template['ingested_in_graph'] = values
+            if values == 'false':
+                failed_deps_count['all'] = failed_deps_count['all'] + 1
+                failed_deps_count[epv_arr[0]] = failed_deps_count[epv_arr[0]] + 1
+            all_epv_list_v2[epv_arr[0]][epv_arr[1]]['versions'].append(versn_template)
+
+        logger.error(graph_input)
```

Comment:
what does this error message denote here?

Reply:
That was for debugging. Removing it.